universal-application-tool-0.0.1/app/modules/SecurityModule.java

```diff
@@ -101,175 +95,72 @@ protected void configure() {
     PlayCookieSessionStore sessionStore =
         new PlayCookieSessionStore(new ShiroAesDataEncrypter(aesKey));
     bind(SessionStore.class).toInstance(sessionStore);
-  }
 
-  @Provides
-  @Singleton
-  protected GuestClient guestClient(ProfileFactory profileFactory) {
-    return new GuestClient(profileFactory);
+    // Default to these options
+    String applicantAuthClient = "idcs";
+
+    try {
+      applicantAuthClient = configuration.getString("auth.applicant_idp");
+    } catch (ConfigurationException error) {
+      // Do nothing and default to adfs.
+    }
+
+    bindAdminIdpProvider();
+    bindApplicantIdpProvider(applicantAuthClient);
   }
 
-  @Provides
-  @Singleton
-  protected FakeAdminClient fakeAdminClient(ProfileFactory profileFactory) {
-    return new FakeAdminClient(profileFactory, this.configuration);
+  private void bindAdminIdpProvider() {
+    // Currently the only supported admin auth provider. As we add other admin auth providers,
+    // this can be converted into a switch statement.
+    bind(IndirectClient.class)
+        .annotatedWith(AdminAuthClient.class)
+        .toProvider(AdOidcProvider.class);
   }
 
-  /** Creates a singleton object of OidcClient configured for IDCS and initializes it on startup. */
-  @Provides
-  @Nullable
-  @Singleton
-  @IdcsOidcClient
-  protected OidcClient provideIDCSClient(
-      ProfileFactory profileFactory, Provider<UserRepository> applicantRepositoryProvider) {
-    if (!this.configuration.hasPath("idcs.client_id")
-        || !this.configuration.hasPath("idcs.secret")) {
-      return null;
-    }
-    OidcConfiguration config = new OidcConfiguration();
-    config.setClientId(this.configuration.getString("idcs.client_id"));
-    config.setSecret(this.configuration.getString("idcs.secret"));
-    config.setDiscoveryURI(this.configuration.getString("idcs.discovery_uri"));
-    config.setResponseMode("form_post");
-    // Our local fake IDCS doesn't support 'token' auth.
-    if (baseUrl.contains("localhost:")) {
-      config.setResponseType("id_token");
-    } else {
-      config.setResponseType("id_token token");
+  private void bindApplicantIdpProvider(String applicantIdpName) {
+    AuthIdentityProviderName idpName = AuthIdentityProviderName.forString(applicantIdpName).get();
+
+    switch (idpName) {
+      case LOGIN_RADIUS_APPLICANT:
+        bind(IndirectClient.class)
+            .annotatedWith(ApplicantAuthClient.class)
+            .toProvider(LoginRadiusSamlProvider.class);
+        break;
+      case IDCS_APPLICANT:
+      default:
```

Comment:
Similar to the other comment, I don't know that we should have a default, all options should be equal I think, regardless of the existing state.

Are you concerned that some of these values won't be set in the existing Seattle deployment?

Reply:
Yes, I feel a lot better if the code defaults to the existing prod options because I'd like to remove the possibility of breaking prod as much as possible. This is what we did for file upload as well, and it felt a lot safer to me.